iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/file_lock.hpp

```diff
@@ -78,21 +86,40 @@ class FileLock : public DesignPattern::Creation<FileLock, FileLockError>
     ~FileLock() noexcept;
 
   private:
+    enum class LockOperation : int32_t
+    {
+        LOCK = LOCK_EX | LOCK_NB,
+        UNLOCK = LOCK_UN
+    };
+
     int32_t m_fd{INVALID_FD};
-    FileName_t m_name;
-    PathName_t m_fileLockPath;
+    FilePath_t m_fileLockPath;
 
-    /// @brief c'tor
-    /// @param[in] name of the created file lock in PATH_PREFIX
-    explicit FileLock(const FileName_t& name) noexcept;
+  private:
+    friend class FileLockBuilder;
+    FileLock(const int32_t fileDescriptor, const FilePath_t& path) noexcept;
 
     void invalidate() noexcept;
 
-    cxx::expected<FileLockError> initializeFileLock() noexcept;
-    FileLockError convertErrnoToFileLockError(const int32_t errnum) const noexcept;
+    static FileLockError convertErrnoToFileLockError(const int32_t errnum, const FilePath_t& fileLockPath) noexcept;
     cxx::expected<FileLockError> closeFileDescriptor() noexcept;
+};
 
-    friend class DesignPattern::Creation<FileLock, FileLockError>;
+class FileLockBuilder
+{
+    /// @brief Defines the file name of the lock
+    IOX_BUILDER_PARAMETER(FileLock::FileName_t, name, "")
```

Comment:
Can you extend the documentation and state that `LOCK_FILE_SUFFIX` will be appended to the name